tests/integration/aliases/adopt_into_component/step2/index.ts

```diff
@@ -27,3 +27,16 @@ class Component extends pulumi.ComponentResource {
 }
 // The creation of the component is unchanged.
 const comp2 = new Component("comp2");
+
+// Scenario 3, adopt this resource into a new parent.
+class Component2 extends pulumi.ComponentResource {
+    constructor(name: string, opts?: pulumi.ComponentResourceOptions) {
+        super("my:module:Component2", name, {}, opts);
+    }
+}
+
+// validate that "parent: undefined" means "i didn't have a parent previously"
+const unparentedComponent = new Component2("unparented", {
+    aliases: [{ parent: undefined }],
```

Comment:
I sort of wonder whether we want a more complex examples as well that looks more scematically like what you hit in awsx that motivated changing this - where you want to be able to pass through `parent: opts.parent` and have an `undefined` there mean something different than not passing `parent: ` at all.

Reply:
yes, that seems reasonable.  will add the tests.

Reply:
Done.